src/Aspire.Hosting/Kafka/KafkaBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using System.Text.Json;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+public static class KafkaBuilderExtensions
+{
+    /// <summary>
+    /// Adds a Kafka broker container to the application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port of Kafka broker.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{KafkaContainerResource}"/></returns>
+    public static IResourceBuilder<KafkaContainerResource> AddKafkaBrokerContainer(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    {
+        const int KafkaBrokerPort = 9092;
+        port ??= KafkaBrokerPort;
+        
+        return builder.AddResource(new KafkaContainerResource(name))
+             .WithAnnotation(new ManifestPublishingCallbackAnnotation((json) => json.WriteString("type", "kafka.server.v0")))
+             .WithAnnotation(new ServiceBindingAnnotation(ProtocolType.Tcp, port: port, containerPort: KafkaBrokerPort))
+             .WithAnnotation(new ContainerImageAnnotation
+             {
+                 Image = "confluentinc/confluent-local",
```

Comment:
How does the .NET team evaluate container image defaults like this? How do we know it's safe to use? Is Microsoft signing off on this as trusted? Are there license, support, privacy, import/export implications?

Reply:
The .NET team cannot evaluate the underlying third-party containers for which we have API support for suitability for specific customer requirements.

You should evaluate whichever containers you chose to compose and automate with Aspire to ensure they meet your, your employers or your government’s requirements on security and safety as well as cryptographic regulations and any other regulatory or corporate stands that may apply to your use of the container.

Reply:
@mitchdenny Mitch, my concerns aren’t about the _specific customer requirements_. Rather, I’m asking about some fundamental safeguards. It’s quite unsettling that my concerns seem to have been deliberately dismissed with a legal response.

https://github.com/dotnet/aspire/blob/main/README.md#note-on-containers-used-by-aspire-componentsresources

Reply:
Appreciate the feedback and sorry for the delay in responding. I've been OOF and haven't managed to get around to everything. I'm sorry you feel that we are dismissing your concerns.

When we say the .NET team cannot evaluate the underlying third-party containers, we mean that ultimately because we are not in control of those third-party container build pipelines, we can't attest to what is in them. Its conceptually the same as referencing one of those containers in a Docker Compose file, Kubernetes deployment or similar container orchestrator. We don't distribute the containers (with a few exceptions).

That said, I do think you've raised some good points. For example, perhaps we should consider referencing the containers we depend on via their SHA256 identifier rather than a tag. But there is a negative flipside to that in that lets say we reference a container via its SHA, and that particular version has a vulnerability then not using a floating tag (either latest or something a bit more version/channel orientated) willl mean you don't automatically get patched.

This is a pretty naunced topic so I would encourage you to open up a specific issue for us to discuss rather than side track this PR.